src/HonzaBotner.Discord.Services/Extensions/StringExtensions.cs

```diff
@@ -36,26 +35,29 @@ static StringExtension()
     private static async Task<string> MentionEvaluateAsync(Match match, DiscordGuild? guild)
     {
         // Invalidate ID pings, replacing them with their name (only in guilds)
-        if (guild is not null && match.Groups.TryGetValue("2", out Group? idMention) && idMention.Value != "")
+        if (guild is null || !match.Groups.TryGetValue("2", out Group? idMention) || idMention.Value == "")
         {
-            ulong snowflakeId = ulong.Parse(idMention.Value);
+            return match.Value.Replace("@",
+                "@" + char.ConvertFromUtf32(int.Parse("200b", System.Globalization.NumberStyles.HexNumber)));
+        }
 
-            if (match.Groups.TryGetValue("1", out Group? idType) && idType.Value == "&")
-            {
-                DiscordRole mentionedRole = guild.GetRole(snowflakeId);
-                if (mentionedRole is not null)
-                    return mentionedRole.Name.RemoveDiscordMentions();
-            }
-            else
+        ulong snowflakeId = ulong.Parse(idMention.Value);
+
+        if (match.Groups.TryGetValue("1", out Group? idType) && idType.Value == "&")
+        {
+            DiscordRole mentionedRole = guild.GetRole(snowflakeId);
+            if (mentionedRole is not null)
+                return mentionedRole.Name.RemoveDiscordMentions();
+        }
+        else
+        {
+            try
             {
-                try
-                {
-                    DiscordMember mentionedMember = await guild.GetMemberAsync(snowflakeId);
-                    return mentionedMember.DisplayName.RemoveDiscordMentions();
-                }
-                catch (NotFoundException)
-                {}
+                DiscordMember mentionedMember = await guild.GetMemberAsync(snowflakeId);
+                return mentionedMember.DisplayName.RemoveDiscordMentions();
             }
+            catch (NotFoundException)
+            {}
         }
```

Comment:
I am not sure if this change increases readability. There is also duplicite code now.

Reply:
Github changed it... This review was for the changes in the whole file StringExtensions.cs